community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java

```diff
@@ -389,7 +390,19 @@ public void force() throws IOException
         synchronized ( logFile )
         {
             writer.emptyBufferIntoChannelAndClearIt();
+        }
+        // Force the writer outside of the lock.
+        // This allows multiple threads forcing at the same time to piggy-back onto one another.
```

Comment:
please correct me if i missing something, but there is another path from `TransactionCommittingResponseUnpacker.applyQueuedTransactions` ?